core/src/main/scala/org/scalafmt/util/TokenOps.scala

```diff
@@ -52,13 +54,26 @@ object TokenOps {
     longHash
   }
 
-  def shouldGet2xNewlines(tok: FormatToken): Boolean = {
+  def shouldGet2xNewlines(tok: FormatToken, style: ScalafmtConfig, owners: Token => Tree): Boolean = {
     !isDocstring(tok.left) && {
       val newlines = newlinesBetween(tok.between)
       newlines > 1 || (isDocstring(tok.right) && !tok.left.is[Comment])
+    } || {
+      style.newlines.alwaysBeforeTopLevelStatements &&
+        tok.between.count(_.is[KwNew]) < 2 && isTopLevelStatment(tok.right, owners)
     }
   }
 
+  def isTopLevelStatment(tok: Token, owners: Token => Tree): Boolean = {
+      tok.is[KwObject] ||
+        tok.is[KwDef] ||
+        tok.is[KwCase] ||
+        tok.is[KwPackage] ||
+        tok.is[KwClass] ||
+        owners(tok).parent.exists(_.is[Defn.Class]) ||
```

Comment:
the KwClass is actualy redundant yes, I removed that. Do you mean that also the rest should be changed?